AlphaWallet/Tokens/ViewModels/TokensViewModel.swift

```diff
@@ -172,8 +172,57 @@ class TokensViewModel {
             return nil
         }
     }
+
+    func indexPathArrayForDeletingAt(indexPath current: IndexPath) -> [IndexPath] {
+        let canRemoveCurrentItem: Bool  = item(for: current.row, section: current.section).isRemovable
+        let canRemovePreviousItem: Bool = {
+            current.row > 0 ? item(for: current.row - 1, section: current.section).isRemovable : false
+        }()
+        let canRemoveNextItem: Bool = {
```

Comment:
When can we remove the next item? Seems like there should only be 3 possibilities when we swipe a token:

A. The token can't be removed because it is a native token. Taken care by a check against `TokenObjectOrRpcServerPair.canDelete` or similar
B. There is more than 1 token in the section (group). The token is removed
C. There is exactly 1 token in the section (group). The token as well as the header is removed

Reply:
It’s to determine if the item is the last one in its “section” or not. The current code does not break the tokens into sections but into one large section and uses rpcServers to display the header. `canDelete`
 doesn’t work because the token is hidden and not actually deleted.

Reply:
> The current code does not break the tokens into sections but into one large section and uses rpcServers to display the header. 

Right. But I still don't understand why we ever consider if we can remove the next token (i.e. 0, 1, 1 in the truth table). Can you give the simplest example of how it would look like in the Wallet tab and which row is being swiped?

Actually, I think I am starting to understand your point after writing this out, do you think you can modify the PR to make it clearer or as a last resort, add a comment to the function `indexPathArrayForDeletingAt(indexPath:)`?

Reply:
> canDelete doesn’t work because the token is hidden and not actually deleted. This was the first thing I tried.

Just to be sure: Tokens are only hidden when the user swipes (no matter if the function names calls them hide or delete. It ought to be the same action). We can't delete tokens from the chain nor really delete in the app (we do remove/move the token entries in the Realm database though, so you might be referring to that).

With that in mind, can you help think/describe that again why it doesn't work? And these 2 checks are important:

```
guard !token.isInvalidated else { return false }
if token.contractAddress.sameContract(as: Constants.nativeCryptoAddressInDatabase) {
```

The 1st line is a defensive check against Realm crashing on us, the next line makes sure we don't hide the native token for each chain

Reply:
`canDelete` doesn't work because it does what it says. The property tells you it can't be deleted. We want to remove the token row from the tableview. `canDelete` doesn't tell me if the token row can be removed since it will always return false for those native tokens for each chain. Swiping on Ethereum will never remove it from the Wallet main screen. 

This is based on the behaviour of the original code which allows Ethereum to be removed. If this is not the case, then a new issue should be filed since this issue deals only with crashing while swiping to delete.